diff --git a/dlls/user32/Makefile.in b/dlls/user32/Makefile.in
index 69e1fa2..220710c 100644
--- a/dlls/user32/Makefile.in
+++ b/dlls/user32/Makefile.in
@@ -3,7 +3,7 @@ MODULE    = user32.dll
 IMPORTLIB = user32
 IMPORTS   = $(PNG_PE_LIBS) gdi32 sechost advapi32 kernelbase win32u uuid
 EXTRAINCL = $(PNG_PE_CFLAGS)
-DELAYIMPORTS = imm32 combase
+DELAYIMPORTS = imm32 combase ws2_32
 
 SOURCES = \
 	button.c \
